lib/logs.coffee

```diff
@@ -15,20 +15,158 @@ limitations under the License.
 ###
 
 Promise = require('bluebird')
-logs = require('resin-device-logs')
+find = require('lodash/find')
+flatMap = require('lodash/flatMap')
+max = require('lodash/max')
+pubnubLogs = require('resin-device-logs')
+rSemver = require('resin-semver')
 errors = require('resin-errors')
+moment = require('moment')
+{ EventEmitter } = require('events')
+
+{ findCallback } = require('./util')
+
+API_LOGS_SUPERVISOR_VERSION_RANGE = '>=7.0.0'
+LOGGING_OVERRIDE_VAR = 'RESIN_SUPERVISOR_NATIVE_LOGGER'
+
+API_POLL_INTERVAL = 1000
 
 getLogs = (deps, opts) ->
+	{ pine } = deps
 	configModel = require('./models/config')(deps, opts)
 	deviceModel = require('./models/device')(deps, opts)
 
 	exports = {}
 
 	getContext = (uuidOrId) ->
 		return Promise.props
-			device: deviceModel.get(uuidOrId)
+			device: deviceModel.get uuidOrId,
+				expand:
+					# Get only the most recent device log
+					owns__device_log:
+						$top: '1'
+						$select: 'device_timestamp'
+						$orderby: 'device_timestamp desc'
+					image_install:
+						$select: 'id'
+						$expand:
+							# Get only the most recent image install log
+							owns__image_install_log:
+								$top: '1'
+								$select: 'device_timestamp'
+								$orderby: 'device_timestamp desc'
+					device_config_variable: {}
+					belongs_to__application:
+						$select: 'id'
+						$expand:
+							application_config_variable: {}
+
 			pubNubKeys: configModel.getPubNubKeys()
 
+	usesApiLogs = (device) ->
+		# If don't know the supervisor, assume it's a new one to start with
+		hasNewSupervisor = !device.supervisor_version or rSemver.satisfies(
+			device.supervisor_version,
+			API_LOGS_SUPERVISOR_VERSION_RANGE
+		)
+
+		if !hasNewSupervisor
+			return false
+
+		deviceVar = find(device.device_config_variable, { name: LOGGING_OVERRIDE_VAR })
```

Comment:
Should the check for the `LOGGING_OVERRIDE_VAR` come before the super version check?

Reply:
I don't think so. The override var is intended only to let you forcibly turn _off_ native logging (mainly so we can disable it if we have perf problems), so only matters for new supervisors.

There's no way to do the opposite (turn it on for old supervisors) so if you've got an old supervisor then you're always using pubnub, while if you've got a new supervisor you're using the API by default, unless the `RESIN_SUPERVISOR_NATIVE_LOGGER` is set to 'false'.